### PR TITLE
Added Xojo Lexer

### PR DESCRIPTION
**Pull request by jimmckay**

### CODE REVIEW

File: lib/rouge/lexers/xojo.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      end
+
+      state :dotted do
+        mixin :whitespace
+        rule id, Name, :pop!
+      end
+
+      state :string do
+        rule /""/, Str::Escape
+        rule /"C?/, Str, :pop!
```
Comment (pyrmont) [inline lib/rouge/lexers/xojo.rb]: Is this correct? Can a string end in a `"C`?

File: lib/rouge/lexers/xojo.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+      state :whitespace do
+        rule /\s+/, Text
+        rule /\n/, Text, :bol
+        rule /rem\b.*?$/i, Comment::Single
+        rule /\/\/.*$/, Comment::Single
+        rule /\#tag Note.*\#tag EndNote/m, Comment::Doc
+      end
+
+      state :bol do
```
Comment (pyrmont) [inline lib/rouge/lexers/xojo.rb]: Two points:

1. This state should probably have a different name. The `:bol` state is typically used by lexers for a state that may occur at the beginning, and only at the beginning, of a file. For example, preprocessor commands in Swift:
  https://github.com/rouge-ruby/rouge/blob/04ce735a0459d1f5ed7301295a7af9b0823e3e64/lib/rouge/lexers/swift.rb#L35-L42

2. Should the `Name::Attribute` rule even be here? It feels like something that should be part of the `:root` state.

File: lib/rouge/lexers/xojo.rb [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          [#]tag.*\n
+        )ix, Name::Tag
+        rule /[.]/, Punctuation, :dotted
+        rule /[(){}!#,:]/, Punctuation
+        rule /End\b/i, Keyword, :end
+        rule /(Dim|Const|Static)\b/i, Keyword, :dim
+        rule /(Function|Sub|Property),?\s?\b/i, Keyword, :funcname
+        rule /(Class|Structure|Enum)\b/i, Keyword, :classname
+
+        rule upper_id do |m|
```
Comment (pyrmont) [inline lib/rouge/lexers/xojo.rb]: Are the keywords always expressed in uppercase? (There don't appear to be any examples in the visual sample.) If keywords are always in uppercase, why not define them that way and then match them as rules? This is how it's done with the Ruby lexer:

https://github.com/rouge-ruby/rouge/blob/04ce735a0459d1f5ed7301295a7af9b0823e3e64/lib/rouge/lexers/ruby.rb#L184

**Comment by jimmckay** [comment]
Had to create a new branch. I'll make a new pull request on that branch.

**Comment by pyrmont** [comment]
This was resubmitted as #1131.

**Pull request closed without merging**